src/main.py

```diff
@@ -2,59 +2,152 @@
+    then these controllers will be mocked controllers that run a simulation.
+    """
+
+    # TODO: Can this fail? If so, make it not.
+    # TODO: In test mode this should be fake.
+    rocket_data = RocketData()
+
+    # TODO: For the GpioReader class, this is where a fake one would be initted.
+
+    # TODO: !MC - Suborbit should be initialized in here.
+    # TODO: Actually initialize everything. (Like gpio reader)
+
+    return (rocket_data)
+
+
+def standby():
+    """Sitting on the rail, waiting for launch.
+
+    The rocket may be in this state for up to 6 hours, and will not get a
+    heads up before the rocket launches.
+
+    This state should be looking and waiting for detection of motors being
+    fired. We will have to detect this through means other than a direct
+    signal (Something like a large acceleration spike).
+
+    Once launch is detected, this method will return.
+    """
+    # TODO: Implement this state
+    pass
+
+
+def powered_flight():
+    """This is when the rocket motor is actively firing.
+
+    This should only last around three seconds.
+
+    The flight computer should stay in this state until either we detect motor
+    burnout, or a manual limit is reached (such as 5s)
+
+    During this stage there should be **no airbrakes**
+    """
+    # TODO: Implement this state
+    pass
+
+
+def coast_flight():
+    """This is when the rocket is still moving upwards to apogee, but after
+    motor burnout.
+
+    This phase will last roughly 27s.
 
-    
-    running = True 
-      
-    if(test_mode): 
-        while(running):
-            #retrieves random data from GPIOReader
-            gpio = GPIOReader(test_mode, False, False, False, False, False, False, False)
-            gpio.retrieveData() 
-            
-            ## Update rocketData object 
+    The flight computer should stay in this state until either we detect apogee,
+    or a manual limit is reached (such as 45s).
 
-            # rocket_data = gpio.retrieveData() TODO: uncomment when X1-AV-43 is merged
+    This state will be responsible for Airbrakes!
+    """
+    # TODO: Implement this state
+    pass
 
-            ## Do airbrakes 
-            
-            ## Send rocketData to datahandler call send function
 
-    else:
-        
-        while(running):
-        ## Setup a loop that continues until we tell it to stop
-            ## Read GPIO Input
-            ## Update rocketData object
-            
-            # rocket_data = gpio.retrieveData() TODO: uncomment when X1-AV-43 is merged
+def recovery():
+    """This is the final state after apogee.
 
-            ## Send GPIO rocketData to datahandler for output
+    This phase is of unknown length (to Mateo. Should ask around to confirm.).
 
-            ## Do air brakes math (to be implemented later with kris)
+    The flight computer should stay in this state until recovered. This is
+    forever as far as the FC is concerned.
```

Comment:
In my head I was thinking the screw switch would be to just turn the whole thing off. This might be something to discuss at the meeting. Having a separate one just to tell the computer to stop recording and clean up all data and connection to the BB might be something worth including.

Reply:
Summary of discussion:

- We should have a way of detecting end of flight so that we can stop logging and clean up open files and connections.
- I will create an issue documenting this.
- Could be we turn off the altimiter first and let the pi detect that.
- Could also be detect an altitude close to launch and wait 5 minutes before turning off.